apps/src/lab2/progress/ProgressManager.ts

```diff
@@ -96,16 +96,21 @@ export default class ProgressManager {
     this.onProgressChange();
   }
 
-  private resetValidation(hasConditions: boolean) {
+  resetValidation() {
     if (this.validator) {
       // Give the lab the chance to clear accumulated satisfied conditions.
       this.validator.clear();
     }
 
+    const hasConditions =
+      (this.levelData?.validations && this.levelData.validations.length > 0) ||
+      false;
+
     this.currentValidationState = {
       hasConditions,
       satisfied: false,
       message: null,
+      index: this.currentValidationState.index + 1,
```

Comment:
Mind adding a comment on why we increment the index here?

Reply:
Good call; added.